File: ios/ObjectiveC/Beacon/8-Beacon-Range/Starter/BeaconRange/ViewController.m
//
//  ViewController.m
//  QuickStart
//
//  Created by sarra srairi on 22/03/2016.
//  Copyright © 2016 sarra srairi. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () {
    NSString *feedStatusString;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void) viewDidAppear:(BOOL)animated{
    // Register Range Delegate protocol to your view
    [super viewDidAppear:animated];
}

- (void) viewDidDisappear:(BOOL)animated{
    [super viewDidDisappear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: ios/SWIFT/Beacon/1-QuickStart/Starter/QuickStart/ViewController.swift
//
//  ViewController.swift
//  QuickStart
//
//  Created by sarra srairi on 10/08/2016.
//  Copyright © 2016 R&D connecthings. All rights reserved.
//
import UIKit
import AdtagConnection
import ConnectPlaceActions
import AdtagLocationBeacon
import ConnectPlaceCommon

class ViewController: UIViewController, AdtagInProximityInForegroundDelegate, ProximityErrorDelegate {
    
    @IBOutlet weak var txt_message: UILabel!
    var adtagInitializer: AdtagInitializer?
    var adtagBeaconManager: AdtagBeaconManager?

    override func viewDidLoad() {
        super.viewDidLoad()
        adtagInitializer = AdtagInitializer.shared
        adtagBeaconManager = AdtagBeaconManager.shared
        // Do any additional setup after loading the view, typically from a nib

    }

    override func viewWillAppear(_ animated: Bool) {
        adtagInitializer?.registerProximityErrorDelegate(self)
        adtagBeaconManager?.registerInProximityInForeground(self)
        super.viewWillAppear(animated)
    }

    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        adtagInitializer?.unregisterProximityErrorDelegate(self)
        adtagBeaconManager?.unregisterInProximityInForeground(self)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func proximityContentsInForeground(contents: [AdtagPlaceInAppAction]) {
        self.txt_message.text = String( format: NSLocalizedString("beaconAround", comment:""), contents.count)
    }

    func onProximityError(errorType: Int, message: NSString) {
        self.txt_message.text = String( format: NSLocalizedString("error", comment:""), errorType, message)
    }
 
}
